README.md

```diff
@@ -250,175 +243,232 @@ class Post
-]);
-create(Post::class, [
-    // Factory objects are persisted before calling Post::setCategory()
-    'category' => factory(Category::class, ['name' => 'My Category']),
-]);
+use App\Entity\Post;
+use Zenstruck\Foundry\ModelFactory;
 
-// OneToMany
-create(Category::class, [
-    'posts' => [
-        $post, // $post is instance of Post, Category::addPost($post) will be called during instantiation
+final class PostFactory extends ModelFactory
+{
+    // ...
 
-        // Proxy objects are converted to object before calling Category::addPost()
-        create(Post::class, ['title' => 'Post B', 'body' => 'body']),
+    public function published(): self
+    {
+        return $this->addState(['published_at' => self::faker()->dateTime]);
+    }
 
-        // Factory objects are persisted before calling Category::addPost()
-        factory(Post::class, ['title' => 'Post A', 'body' => 'body']),
-    ],
-]);
+    public function unpublished(): self
+    {
+        return $this->addState(['published_at' => null]);
+    }
 
-// ManyToMany
-create(Post::class, [
-    'tags' => [
-        $tag, // $tag is instance of Tag, Post::addTag($tag) will be called during instantiation
+    public function withViewCount(int $count = null): self
+    {
+        return $this->addState(function () use ($count) {
+            return ['view_count' => $count ?? self::faker()->numberBetween(0, 10000)];
+        });
+    }
+}
+```
 
-        // Proxy objects are converted to object before calling Post::addTag()
-        create(Tag::class, ['name' => 'My Tag']),
+You can use states to make your tests very explicit to improve readability:
 
-        // Factory objects are persisted before calling Post::addTag()
-        factory(Tag::class, ['name' => 'My Tag']),
-    ],
-]);
+```php
+$post = PostFactory::new()->unpublished()->create();
+$post = PostFactory::new()->withViewCount(3)->create();
+
+// combine multiple states
+$post = PostFactory::new()
+    ->unpublished()
+    ->withViewCount(10)
+    ->create()
+;
+
+// states that don't require arguments can be added as strings to PostFactory::new()
+$post = PostFactory::new('published', 'withViewCount')->create();
+```
+
+### Attributes
+
+The attributes used to instantiate the object can be added several ways. Attributes can be an *array*, or a *callable*
+that returns an array. Using a *callable* ensures random data as the callable is run for each object separately during
+instantiation.
+
+```php
+use App\Entity\Category;
+use App\Entity\Post;
+use App\Factory\CategoryFactory;
+use App\Factory\PostFactory;
+use function Zenstruck\Foundry\faker;
+
+$post = PostFactory::new(['title' => 'Post A'])
+    ->withAttributes([
+        'body' => 'Post A Body...',
+
+        // can use snake case
+        'published_at' => new \DateTime('now'), 
+
+        // factories are automatically instantiated (and persisted if the outer factory is persisted)
+        'category' => CategoryFactory::new(['name' => 'php']), 
```

Comment:
No, in this case, each post created would get a different category because you passed an instance of the `CategoryFactory` not the `Category` itself.

`CategoryFactory::new(['name' => 'php'])->create()` would create a new category and use that for each post (unless your attributes are a `callable`... then it would be a new one for each post).